src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.ItemGetter.cs

```diff
@@ -0,0 +1,280 @@
+                {
+                    return [];
+                }
+
+                var targetToken = SyntaxTree
+                    .FindTokenOnLeftOfPosition(Position, CancellationToken)
+                    .GetPreviousTokenIfTouchingWord(Position);
+
+                if (!SyntaxTree.IsRightOfDotOrArrowOrColonColon(Position, targetToken, CancellationToken))
+                    return [];
+
+                var node = targetToken.Parent;
+                // Bind the interface name which is to the left of the dot
+                NameSyntax? name = null;
+                switch (node)
+                {
+                    case ExplicitInterfaceSpecifierSyntax specifierNode:
+                        name = specifierNode.Name;
+                        break;
+
+                    case QualifiedNameSyntax qualifiedName
+                    when node.Parent.IsKind(SyntaxKind.IncompleteMember):
+                        name = qualifiedName.Left;
+                        break;
+
+                    default:
+                        return [];
+                }
+
+                var semanticModel = await Document.ReuseExistingSpeculativeModelAsync(Position, CancellationToken).ConfigureAwait(false);
+                var symbol = semanticModel.GetSymbolInfo(name, CancellationToken).Symbol as ITypeSymbol;
+                if (symbol?.TypeKind != TypeKind.Interface)
+                    return [];
+
+                var typeDeclaration = node.GetAncestor<BaseTypeDeclarationSyntax>();
+                if (typeDeclaration is null)
+                    return [];
+
+                var containingType = semanticModel.GetDeclaredSymbol(typeDeclaration, CancellationToken);
+                if (containingType is null)
+                    return [];
+
+                // We must be explicitly implementing the interface
+                if (!containingType.Interfaces.Contains(symbol))
+                    return [];
+
+                // We're going to create a entry for each one, including the signature
+                var namePosition = name.SpanStart;
+                var text = await Document.GetValueTextAsync(CancellationToken).ConfigureAwait(false);
+                text.GetLineAndOffset(namePosition, out var line, out var lineOffset);
+                var items = symbol.GetMembers()
+                    .Where(FilterInterfaceMember)
+                    .SelectAsArray(CreateCompletionItem);
+
+                return items;
+
+                bool FilterInterfaceMember(ISymbol member)
+                {
+                    if (!member.IsAbstract && !member.IsVirtual)
+                        return false;
+
+                    if (member.IsAccessor() ||
+                        member.Kind == SymbolKind.NamedType ||
+                        !semanticModel.IsAccessible(node.SpanStart, member))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                }
```

Comment:
all off these need docs.  and docs on 'why' not 'what' :)

Reply:
Docs were placed, please check